src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -0,0 +1,199 @@
+        // is thrown
+        // 
+        public Exception(string message, Exception innerException)
+            : this()
+        {
+            _message = message;
+            _innerException = innerException;
+        }
+
+        protected Exception(SerializationInfo info, StreamingContext context)
+        {
+            if (info == null)
+                throw new ArgumentNullException(nameof(info));
+
+            _className = info.GetString("ClassName"); // Do not rename (binary serialization)
+            _message = info.GetString("Message"); // Do not rename (binary serialization)
+            _data = (IDictionary)(info.GetValueNoThrow("Data", typeof(IDictionary))); // Do not rename (binary serialization)
+            _innerException = (Exception)(info.GetValue("InnerException", typeof(Exception))); // Do not rename (binary serialization)
+            _helpURL = info.GetString("HelpURL"); // Do not rename (binary serialization)
+            _stackTraceString = info.GetString("StackTraceString"); // Do not rename (binary serialization)
+            _HResult = info.GetInt32("HResult"); // Do not rename (binary serialization)
+            _source = info.GetString("Source"); // Do not rename (binary serialization)
+
+            RestoreRemoteStackTrace(info, context);
+        }
+
+        public virtual string Message
+        {
+            get
+            {
+                if (_message == null)
+                {
+                    return SR.Format(SR.Exception_WasThrown, GetClassName());
+                }
+                else
+                {
+                    return _message;
+                }
+            }
+        }
+
+        public virtual IDictionary Data
+        {
+            get
+            {
+                if (_data == null)
+                    _data = CreateDataContainer();
+
+                return _data;
+            }
+        }
+
+        private string GetClassName()
+        {
+            // Will include namespace but not full instantiation and assembly name.
+            if (_className == null)
+                _className = GetType().ToString();
+
+            return _className;
+        }
+
+        // Retrieves the lowest exception (inner most) for the given Exception.
+        // This will traverse exceptions using the innerException property.
+        //
+        public virtual Exception GetBaseException()
+        {
+            Exception inner = InnerException;
+            Exception back = this;
+
+            while (inner != null)
+            {
+                back = inner;
+                inner = inner.InnerException;
+            }
+
+            return back;
+        }
+
+        public Exception InnerException => _innerException;
+
+        // Sets the help link for this exception.
+        // This should be in a URL/URN form, such as:
+        // "file:///C:/Applications/Bazzal/help.html#ErrorNum42"
+        // Changed to be a read-write String and not return an exception
```

Comment:
"Changed to be a read-write String and not return an exception" comment is not very useful.